app/code/Magento/Sales/Observer/SalesEventQuoteSubmitBeforeObserver.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Sales\Observer;
+
+use Magento\Framework\Event\ObserverInterface;
+
+/**
+ * Copy custom fields from quote tables to order.
+ */
+class SalesEventQuoteSubmitBeforeObserver implements ObserverInterface
+{
+    /**
+     * @var \Magento\Framework\DataObject\Copy\Config
+     */
+    private $fieldsetConfig;
+
+    /**
+     * SalesEventQuoteSubmitBeforeObserver constructor.
+     *
+     * @param \Magento\Framework\DataObject\Copy\Config $fieldsetConfig
+     */
+    public function __construct(
+        \Magento\Framework\DataObject\Copy\Config $fieldsetConfig
+    ) {
+        $this->fieldsetConfig = $fieldsetConfig;
+    }
+
+    /**
+     * @param \Magento\Framework\Event\Observer $observer
+     *
+     * @return $this
+     */
+    public function execute(\Magento\Framework\Event\Observer $observer)
+    {
+        /** @var \Magento\Quote\Api\Data\CartInterface $quote */
+        $quote = $observer->getEvent()->getQuote();
+
+        /** @var \Magento\Sales\Api\Data\OrderInterface $order */
+        $order = $observer->getEvent()->getOrder();
+
+        $fields = $this->fieldsetConfig->getFieldset('sales_convert_quote', 'global');
+
+        $methods = get_class_methods($order);
```

Comment:
The `\Magento\Framework\Api\DataObjectHelper::mergeDataObjects()` reflects methods of `\Magento\Sales\Api\Data\OrderInterface` and here methods of the implementation of that interface are used. If you have set a different preference for the `OrderInterface` and have added getter methods in that implementation, these will not be included int both copy actions: not in the one done by the DataObjectHelper and not in the one here.

You could use `\Magento\Framework\Reflection\MethodsMap::getMethodsMap(OrderInterface::class)` like is being used by the DataObjectHelper.